src/Spectre.Angular2Client/src/app/spectrums/spectrum/spectrum.component.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * spectrum.component.ts
+ * Component of single spectrum.
+ *
+ Copyright 2017 Grzegorz Mrukwa, Sebastian Pustelnik, Daniel Babiak
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import {Component, Input, OnInit} from '@angular/core';
+
+@Component({
+  selector: 'app-spectrum',
+  templateUrl: './spectrum.component.html',
+  styleUrls: ['./spectrum.component.css']
+})
+
+export class SpectrumComponent implements OnInit {
+  @Input() public spectrumData: any;
```

Comment:
Also here we have `any`. If we moved to well-typed input (i.e. `Spectrum`) and translate it internally to Plotly-understandable format, it would be more useful from the outside.